client/src/actions/index.js

```diff
@@ -0,0 +1,15 @@
+export const FETCH_PRODUCTS = 'FETCH_PRODUCTS';
+const url = `http://localhost:8000/products`;
+
+export const fetchProducts = async query => {
```

Comment:
For this action, I'd suggest using a thunk ([redux-thunk](https://codepen.io/stowball/post/a-dummy-s-guide-to-redux-and-thunk-in-react)) middleware, because by default Redux action creators don't support asynchronous actions like fetching data, so here's where we utilize Redux Thunk. Thunk allows you to write action creators that return a function instead of an action. This action, as it is, is leading to problems when trying to categorize or sort your products.